brain/circuit.h

```diff
@@ -61,15 +61,24 @@ class Circuit : public boost::noncopyable
     BRAIN_API  ~Circuit();
 
     /**
-     * @return The set of GIDs for the given target name. If empty it will
-     *         return all the GIDs held by the circuit. If the target cannot be
-     *         found, a runtime exception is raised.
+     * @return The set of GIDs for the given target name.
+     * @throw std::runtime_error if the target cannot be found.
      */
     BRAIN_API GIDSet getGIDs( const std::string& target ) const;
 
     /** @return All GIDs held by the circuit */
     BRAIN_API GIDSet getGIDs() const;
 
+    /**
+     * @return A random fraction of GIDs from the given target name.
+     * @throw std::runtime_error if the target cannot be found.
+     */
+    BRAIN_API GIDSet getRandomGIDs( float fraction,
+                                    const std::string& target ) const;
+
+    /** @return A random fraction of GIDs from the circuit. */
+    BRAIN_API GIDSet getRandomGIDs( float fraction ) const;
```

Comment:
In the implementation you're reading the whole target first. Why not make this functions free functions then:

    randomSampling(circuit.getGIDs(), 0.1)